### PR TITLE
Add Example DSL

### PR DESCRIPTION
**Pull request by DavyC**
Add Example DSL to allow users define their custom example, so goa can generate swagger specs with examples. If not given, goa will automatically generate a random value for the users.

### CODE REVIEW

File: design/types.go [hunk 1]
```diff
@@ -133,6 +133,8 @@ const (
 	UserTypeKind
 	// MediaTypeKind represents a media type.
 	MediaTypeKind
+	// NoneKind represents a nil
```
Comment (raphael) [inline design/types.go]: Is that really needed? adding a new type has a lot of consequences... all generators will have to be updated.

File: design/apidsl/media_type_test.go [hunk 2]
```diff
@@ -296,3 +296,203 @@ var _ = Describe("CollectionOf", func() {
 		})
 	})
 })
+
```
Comment (raphael) [inline design/apidsl/media_type_test.go]: Can you add a tests with cyclical dependencies? something like
```go
var mt = MediaType("vnd.application/foo", func() {
    Attributes(func() {
        Attribute("foo", "vnd.application/bar")
    })
})

var mt2 = MediaType("vnd.application/bar", func() {
    Attributes(func() {
        Attribute("bar", mt)
    })
})
```

File: design/types.go (hunk 1 above)
Comment (DavyC) [inline design/types.go]: I think I am leaning toward `NoExample()` in the end. The `NoneKind` seems a bit too much for one purpose I agree.

Comment (raphael) [inline design/types.go]: Great, thanks!

**Comment by raphael** [comment]
This is awsome, thank you!

**Pull request merged**